backend/src/main/java/com/capstone/moneytree/WebSocketConfig.java

```diff
@@ -0,0 +1,35 @@
+package com.capstone.moneytree;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.simp.config.MessageBrokerRegistry;
+import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker;
+import org.springframework.web.socket.config.annotation.StompEndpointRegistry;
+import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer;
+
+@Configuration
+@EnableWebSocketMessageBroker
+public class WebSocketConfig implements WebSocketMessageBrokerConfigurer {
+
+    /**
+     * Registers the asynchronous endpoint websockets messages
+     * withSockJS is enabled for fallback options if websockets are not available in the browser
+     * @param registry The STOMP endpoint registry
+     */
+    @Override
+    public void registerStompEndpoints(StompEndpointRegistry registry) {
+        registry.addEndpoint("/ws")
+                .setAllowedOrigins("http://localhost:4200", "http://127.0.0.1")
```

Comment:
For this line, is there a way to have a variable so it's good for our UAT / PROD environment as well?

Reply:
No need for that. Since both frontend and backend are on the same network and that the backend only gets request from the frontend (both locally on any environment) then we use localhost:4200 for when we test locally and 127.0.0.1 in production. We did the same thing for the MoneyTreeController before. You can have a look at that file to compare.